src/ColinHDev/CPlot/provider/DataProvider.php

```diff
@@ -1291,4 +1306,255 @@ public function savePlotRate(Plot $plot, PlotRate $plotRate) : Generator {
+                            "schema" => $sqlSettings["DatabaseName"],
+                            "port" => $sqlSettings["Port"]
+                        ],
+                        "worker-limit" => ResourceManager::getInstance()->getConfig()->getNested("database.worker-limit", 1)
+                    ], [
+                        "mysql" => "sql" . DIRECTORY_SEPARATOR . "myplot_mysql.sql"
+                    ]);
+                    $records = yield from $myplotDatabase->asyncSelect(self::EXPORT_MYPLOT_PLOTS);
+                    $mergeRecords = yield from $myplotDatabase->asyncSelect(self::EXPORT_MYPLOT_MERGES);
+                    break;
+                case 'yaml':
+                    $filename = "plots.yml";
+                case 'json':
+                    $data = new Config(Path::join(Server::getInstance()->getDataPath(), "plugin_data", "MyPlot", "Data", $filename ?? "plots.json"), Config::DETECT);
+                    $records = array_values($data->get("plots"));
+                    $unparsedMergeRecords = $data->get("merges");
+                    $mergeRecords = [];
+                    foreach($unparsedMergeRecords as $origin => $merges) {
+                        $originData = explode(";", $origin);
+                        foreach($merges as $merge) {
+                            $mergeData = explode(";", $merge);
+                            $mergeRecords[] = [
+                                "level" => $originData[0],
+                                "originX" => $originData[1],
+                                "originZ" => $originData[2],
+                                "mergedX" => $mergeData[0],
+                                "mergedZ" => $mergeData[1]
+                            ];
+                        }
+                    }
+                    break;
+                default:
+                    return; // don't import anything due to invalid data provider
+            }
+            foreach($records as $record) {
+                // validate offline player data
+                $UUID = $XUID = null;
+                $offlineData = Server::getInstance()->getOfflinePlayerData($record["owner"]);
+                if($offlineData !== null) {
+                    $UUID = $XUID = $offlineData->getString("LastKnownXUID", "");
+                    if($UUID === "") {
+                        $skinTag = $offlineData->getCompoundTag("Skin");
+                        $skinData = $skinTag->getByteArray("Data");
+                        $UUID = Uuid::uuid3(Uuid::NIL, ((string)Entity::nextRuntimeId()) . $skinData . $record["owner"]);
+                    }else{
+                        $UUID = Uuid::fromInteger($UUID);
+                    }
+
+                    // register player data
+                    yield from $this->updatePlayerData(
+                        $UUID->getBytes(), // doesn't matter what is input at this point. will overwrite on login
```

Comment:
We probably should just use `null` here instead of storing wrong data.

Reply:
updatePlayerData doesn't accept a null input

Reply:
My bad.
It only does on the `development` branch. I never merged it into this one.